lambdas/data-migration1/src/collections.ts

```diff
@@ -0,0 +1,131 @@
+import { envUtils } from '@cumulus/common';
+import Logger from '@cumulus/logger';
+
+import { RecordAlreadyMigrated } from './errors';
+import { MigrationSummary } from './types';
+
+const Manager = require('@cumulus/api/models/base');
+const schemas = require('@cumulus/api/models/schemas');
+
+const logger = new Logger({ sender: '@cumulus/data-migration/collections' });
+
+export interface RDSCollectionRecord {
+  name: string
+  version: string
+  process: string
+  granuleIdValidationRegex: string
+  granuleIdExtractionRegex: string
+  files: string
+  // default will be set by schema validation
+  duplicateHandling: string
+  // default will be set by schema validation
+  reportToEms: boolean
+  sampleFileName?: string
+  url_path?: string
+  ignoreFilesConfigForDiscovery?: boolean
+  meta?: object
+  tags?: string
+  created_at: Date
+  updated_at: Date
+}
+
+/**
+ * Migrate collection record from Dynamo to RDS.
+ *
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} dynamoRecord
+ *   Source record from DynamoDB
+ * @param {Knex} knex - Knex client for writing to RDS database
+ * @returns {Promise<number>} - Cumulus ID for record
+ * @throws {RecordAlreadyMigrated}
+ *   if record was already migrated
+ */
+export const migrateCollectionRecord = async (
+  dynamoRecord: AWS.DynamoDB.DocumentClient.AttributeMap,
+  knex: Knex
+): Promise<void> => {
+  // Use API model schema to validate record before processing
+  Manager.recordIsValid(dynamoRecord, schemas.collection);
+
+  const existingRecord = await knex<RDSCollectionRecord>('collections')
+    .where({
+      name: dynamoRecord.name,
+      version: dynamoRecord.version,
+    })
+    .first();
+  // Throw error if it was already migrated.
+  if (existingRecord) {
+    throw new RecordAlreadyMigrated(`Collection name ${dynamoRecord.name}, version ${dynamoRecord.version} was already migrated, skipping`);
+  }
+
+  // Map old record to new schema.
+  const updatedRecord: RDSCollectionRecord = {
+    name: dynamoRecord.name,
+    version: dynamoRecord.version,
+    process: dynamoRecord.process,
+    url_path: dynamoRecord.url_path,
+    duplicateHandling: dynamoRecord.duplicateHandling,
+    granuleIdValidationRegex: dynamoRecord.granuleId,
+    granuleIdExtractionRegex: dynamoRecord.granuleIdExtraction,
+    // have to stringify on an array of values
+    files: JSON.stringify(dynamoRecord.files),
+    reportToEms: dynamoRecord.reportToEms,
+    sampleFileName: dynamoRecord.sampleFileName,
+    ignoreFilesConfigForDiscovery: dynamoRecord.ignoreFilesConfigForDiscovery,
+    meta: dynamoRecord.meta ? dynamoRecord.meta : undefined,
+    // have to stringify on an array of values
+    tags: dynamoRecord.tags ? JSON.stringify(dynamoRecord.tags) : undefined,
+    created_at: new Date(dynamoRecord.createdAt),
+    updated_at: new Date(dynamoRecord.updatedAt),
+  };
+
+  return knex('collections').insert(updatedRecord);
```

Comment:
This is returning a value despite the return type of `Promise<void>`

```suggestion
  await knex('collections').insert(updatedRecord);
```

Reply:
ae2047d5a0498188fdf241e81466c2ee176f0158